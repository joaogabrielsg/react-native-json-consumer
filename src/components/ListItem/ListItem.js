import React from 'react';
import { Text, View, StyleSheet } from 'react-native';

const ListItem = ({ data }) => (
  <View style={styles.container}>
    <Text style={styles.text}>id: {data.id}</Text>
    <Text style={styles.text}>name: {data.name}</Text>
    <Text style={styles.text}>pwd: {data.pwd}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#0b478e',
    width: '80%',
    padding: 20,
    borderRadius: 15
  },
  text: {
    textAlign: 'center',
    color: '#ffffff',
    fontSize: 20
  }
});

export default ListItem;
