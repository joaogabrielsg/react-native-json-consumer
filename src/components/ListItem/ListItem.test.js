import 'react-native';
import React from 'react';
import ListItem from './ListItem';

import renderer from 'react-test-renderer';

describe('ListItem', () => {
  it('sholud render correctly', () => {
    const listItem = renderer.create(<ListItem data={{ id: '12', name: 'test', pwd: '123' }} />).toJSON();
    expect(listItem).toMatchSnapshot();
  });
});
