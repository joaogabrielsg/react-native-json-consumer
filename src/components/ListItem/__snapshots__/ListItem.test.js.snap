// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem sholud render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(0,	208, 229)",
      "borderRadius": 12,
      "padding": 20,
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    id: 
    12
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    name: 
    test
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    pwd: 
    123
  </Text>
</View>
`;
